vassal-app/src/main/java/VASSAL/counters/BorderOutline.java

```diff
@@ -0,0 +1,220 @@
+/*
+ *
+ * Copyright (c) 2023 by Vassalengine.org, Brian Reynolds
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License (LGPL) as published by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, copies are available
+ * at http://www.opensource.org.
+ */
+package VASSAL.counters;
+
+import VASSAL.build.module.documentation.HelpFile;
+import VASSAL.command.Command;
+import VASSAL.configure.ColorConfigurer;
+import VASSAL.configure.IntConfigurer;
+import VASSAL.configure.StringConfigurer;
+import VASSAL.i18n.Resources;
+import VASSAL.i18n.TranslatablePiece;
+import VASSAL.tools.SequenceEncoder;
+
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Graphics;
+import java.awt.Rectangle;
+import java.awt.Shape;
+import java.util.Objects;
+
+/**
+ * Trait to draw a colored border around a piece
+ */
+public class BorderOutline extends Decorator implements TranslatablePiece {
+  public static final String ID = "border;"; // NON-NLS
+
+  private String propertyName;
+  private String description;
+  private int thickness = 2;
+  private Color color = Color.RED;
+
+  private final ColoredBorder border = new ColoredBorder();
+
+  public BorderOutline() {
+    this(ID + ";", null); // NON-NLS
+  }
+
+  public BorderOutline(String type, GamePiece p) {
+    mySetType(type);
+    setInner(p);
+  }
+
+  @Override
+  public void mySetType(String type) {
+    final SequenceEncoder.Decoder st = new SequenceEncoder.Decoder(type, ';');
+    st.nextToken();
+    propertyName = st.nextToken("");
+    description  = st.nextToken("");
+    thickness    = st.nextInt(2);
+    color        = st.nextColor(Color.RED);
+
+    border.setColor(color);
+    border.setThickness(thickness);
+  }
+
+  @Override
+  public void mySetState(String newState) {
+  }
+
+  @Override
+  public String myGetState() {
+    return "";
+  }
+
+  @Override
+  public String myGetType() {
+    final SequenceEncoder se = new SequenceEncoder(';');
+    se.append(propertyName).append(description).append(thickness).append(color);
+    return ID + se.getValue();
+  }
+
+  @Override
+  protected KeyCommand[] myGetKeyCommands() {
+    return KeyCommand.NONE;
+  }
+
+  @Override
+  public Command myKeyEvent(javax.swing.KeyStroke stroke) {
+    return null;
+  }
+
+  @Override
+  public Shape getShape() {
+    return piece.getShape();
+  }
+
+  @Override
+  public Rectangle boundingBox() {
+    final Rectangle r = piece.boundingBox();
+    r.add(piece.boundingBox());
```

Comment:
Why are you adding the bounding box when `r` starts out as the bounding box?

Reply:
Because of poor copying and pasting I guess :) All fixed